src/resources/views/filters/text.blade.php

```diff
@@ -0,0 +1,102 @@
+	filter-type="{{ $filter->type }}"
+	class="dropdown {{ Request::get($filter->name)?'active':'' }}">
+	<a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">{{ $filter->label }} <span class="caret"></span></a>
+	<div class="dropdown-menu">
+		<div class="form-group backpack-filter m-b-0">
+			<div class="input-group">
+		        <input class="form-control pull-right"
+		        		id="text-filter-{{ str_slug($filter->name) }}"
+		        		type="text"
+						@if ($filter->currentValue)
+							value="{{ $filter->currentValue }}"
+						@endif
+		        		>
+		        <div class="input-group-addon">
+		          <a class="text-filter-{{ str_slug($filter->name) }}-clear-button" href=""><i class="fa fa-times"></i></a>
+		        </div>
+		    </div>
+		</div>
+	</div>
+</li>
+
+{{-- ########################################### --}}
+{{-- Extra CSS and JS for this particular filter --}}
+
+{{-- FILTERS EXTRA CSS  --}}
+{{-- push things in the after_styles section --}}
+
+@push('crud_list_styles')
+    {{-- <link href="{{ asset('vendor/adminlte/plugins/datepicker/datepicker3.css') }}" rel="stylesheet" type="text/css" /> --}}
+	<style>
+		.input-group.date {
+			width: 320px;
+			max-width: 100%;
+		}
+	</style>
+@endpush
+
+
+{{-- FILTERS EXTRA JS --}}
+{{-- push things in the after_scripts section --}}
+
+@push('crud_list_scripts')
+	<!-- include select2 js-->
+	{{-- <script type="text/javascript" src="{{ asset('vendor/adminlte/plugins/datepicker/bootstrap-datepicker.js') }}"></script> --}}
+  <script>
+		jQuery(document).ready(function($) {
+			$('#text-filter-{{ str_slug($filter->name) }}').on('change', function(e) {
+
+				var parameter = '{{ $filter->name }}';
+				var value = $(this).val();
+
+				@if (!$crud->ajaxTable())
+					// behaviour for normal table
+					var current_url = normalizeAmpersand('{{ Request::fullUrl() }}');
+					var new_url = addOrUpdateUriParameter(current_url, parameter, value);
+
+					// refresh the page to the new_url
+					new_url = normalizeAmpersand(new_url.toString());
+			    	window.location.href = new_url;
+			    @else
+			    	// behaviour for ajax table
+					var ajax_table = $('#crudTable').DataTable();
+					var current_url = ajax_table.ajax.url();
+					var new_url = addOrUpdateUriParameter(current_url, parameter, value);
+
+					// replace the datatables ajax url with new_url and reload it
+					new_url = normalizeAmpersand(new_url.toString());
+					ajax_table.ajax.url(new_url).load();
+
+					// mark this filter as active in the navbar-filters
+					if (URI(new_url).hasQuery('{{ $filter->name }}', true)) {
+						$('li[filter-name={{ $filter->name }}]').removeClass('active').addClass('active');
+					}
+					else
+					{
+						$('li[filter-name={{ $filter->name }}]').trigger('filter:clear');
+					}
+			    @endif
+			});
+
+			$('li[filter-name={{ str_slug($filter->name) }}]').on('filter:clear', function(e) {
```

Comment:
Hmmm, this `str_slug` thing appears quite often:

1. What is it?
2. And it appears it might be something that probably should be in the field options with a default value but that can be overridden...I think.
3. And it's used in a few attributes which means with a bit of fiddling I think someone could come up with aname that just breaks the JavaScript and/or HTML but maybe not

(I don't think they'll make a security concern, although I don't trust Blade as much as Twig in this respect).

Reply:
Yeah... I know I introduced this as best practice a while back, transforming the filter name to a slug. I can't remember _exactly_ why - I bet it was because someone tried to name a filter something obscure and the javascript didn't like it.